File: Programing_Basics_with_JS/whileLoopExercise/07.Moving.js
function moving(input) {
    let index = 0;
    let widthFreeSpace = Number(input[index]);
    index++;
    let lengthFreeSpace = Number(input[index]);
    index++;
    let heightFreeSpace = Number(input[index]);
    index++;

    let totalFreeSpace = widthFreeSpace * lengthFreeSpace * heightFreeSpace;
    let command = input[index];
    index++;

    while(command !== 'Done') {
        let currentLuggage = Number(command);
        totalFreeSpace -= currentLuggage;

        if (totalFreeSpace < 0) {
            break;
        }

        command = input[index];
        index++;
    }

    if (totalFreeSpace >= 0) {
        console.log(`${totalFreeSpace} Cubic meters left.`);
    } else {
        console.log(`No more free space! You need ${Math.abs(totalFreeSpace)} Cubic meters more.`);

    }
}

moving(["10", 
"10",
"2",
"20",
"20",
"20",
"20",
"122"])

moving(["10", 
"1",
"2",
"4", 
"6",
"Done"])

File: Programming_Basics_with_Python/while_loop_exercise/07._moving.py
width_of_free_space = int(input())
length_of_free_space = int(input())
height_of_free_space = int(input())
command = input()

free_space = width_of_free_space * length_of_free_space * height_of_free_space

while not command == "Done":
    num_of_boxes = int(command)
    free_space -= num_of_boxes

    if free_space < 0:
        print(f"No more free space! You need {abs(free_space)} Cubic meters more.")
        break

    command = input()

if free_space >= 0:
    print(f'{free_space} Cubic meters left.')


